File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/MainActivityView.java
package app.editors.manager.mvp.views.main;

import android.graphics.drawable.Drawable;

import androidx.annotation.StringRes;

import com.google.android.play.core.review.ReviewInfo;

import app.editors.manager.app.WebDavApi;
import app.editors.manager.mvp.views.base.BaseViewExt;
import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface MainActivityView extends BaseViewExt {
    void onDialogClose();
    void onRemotePlayMarket(@StringRes int title, @StringRes int info, @StringRes int accept, @StringRes int cancel);
    void onRemoteApp(@StringRes int title, @StringRes int info, @StringRes int accept, @StringRes int cancel);
    void onRatingApp();
    void onAccountAvatar(Drawable resource);
    void onQuestionDialog(String title, String tag, String accept, String cancel, String question);
    void onShowEditMultilineDialog(String title, String hint, String accept, String cancel, String tag);
    void onShowPlayMarket(String releaseId);
    void onShowInAppReview(ReviewInfo mReviewInfo);
    void onShowApp(String releaseId);
    void onShowEmailClientTemplate(String value);
    void onShowOnBoarding();
    void onShowToolbarAccount(String portal, String login, boolean isVisible);
    void onSetWebDavImage(WebDavApi.Providers provider);
    void onShowRecentFragment(boolean isRestore);
    void onShowWebDavFragment(boolean isRestore, WebDavApi.Providers provider);
    void onShowCloudFragment(boolean isRestore, boolean isNoPortal);
    void onShowOnDeviceFragment(boolean isRestore);
    void onShowAccountsFragment(boolean isRestore);
    void onShowProfileFragment();
    void onClearStack();
    void onCloseActionDialog();
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/MainActivityView.kt
package app.editors.manager.mvp.views.main

import android.content.Intent
import android.net.Uri
import androidx.annotation.StringRes
import app.documents.core.model.cloud.Access
import app.editors.manager.mvp.views.base.BaseViewExt
import com.google.android.play.core.review.ReviewInfo
import lib.toolkit.base.managers.utils.EditType
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface MainActivityView : BaseViewExt {
    fun onDialogClose()
    fun onRemotePlayMarket(@StringRes title: Int, @StringRes info: Int, @StringRes accept: Int, @StringRes cancel: Int)
    fun onRemoteApp(@StringRes title: Int, @StringRes info: Int, @StringRes accept: Int, @StringRes cancel: Int)
    fun onRatingApp()
    fun onQuestionDialog(title: String, tag: String, accept: String, cancel: String, question: String?)
    fun onShowEditMultilineDialog(title: String, hint: String, accept: String, cancel: String, tag: String)
    fun onShowPlayMarket(releaseId: String)
    fun onShowInAppReview(reviewInfo: ReviewInfo)
    fun onShowApp(releaseId: String)
    fun onShowEmailClientTemplate(value: String)
    fun onShowOnBoarding()
    fun onCloseActionDialog()
    fun onLocaleConfirmation()
    fun signInAndOpenDeeplink(portal: String, email: String, uri: Uri)
    fun restartActivity(deeplink: Uri? = null)
    fun showEditors(
        uri: Uri,
        editType: EditType,
        access: Access,
        onResultListener: ((Int, Intent?) -> Unit)?
    )
    fun showEditors(
        data: String,
        extension: String,
        editType: EditType,
        access: Access,
        onResultListener: ((Int, Intent?) -> Unit)?
    )
}